src/test/java/io/vertx/httpproxy/ProxyTest.java

```diff
@@ -110,4 +110,56 @@ public Future<Void> handleProxyResponse(ProxyContext context) {
         latch.countDown();
       });
   }
+
+  @Test
+  public void testUpstreamRefuse(TestContext ctx) {
+    SocketAddress backend = SocketAddress.inetSocketAddress(8081, "localhost");
+    startProxy(proxy -> proxy.origin(backend));
+    client = vertx.createHttpClient();
+    Async async = ctx.async();
+    client.request(HttpMethod.GET, 8080, "localhost", "/")
+      .compose(req -> req.send().compose(resp -> {
+        ctx.assertEquals(502, resp.statusCode());
+        return resp.body();
+      }))
+      .onComplete(ctx.asyncAssertSuccess(body -> async.complete()));
+  }
```

Comment:
It seems this test passes without the changes. And I can't see the message "request has already been read" that @patrice-conil described in #75 

Perhaps could you file a separate issue for the problem with interceptor failures?